examples/manual_routes/src/tests.rs

```diff
@@ -0,0 +1,80 @@
+use super::*;
+use rocket::{Rocket, Route, Catcher};
+use rocket::testing::MockRequest;
+use rocket::http::{ContentType, Status};
+use rocket::http::Method::*;
+
+fn route_and_mount() -> Rocket{
+    let always_forward = Route::ranked(1, Get, "/", forward);
+    let hello = Route::ranked(2, Get, "/", hi);
+
+    let echo = Route::new(Get, "/echo:<str>", echo_url);
+    let name = Route::new(Get, "/<name>", name);
+    let post_upload = Route::new(Post, "/", upload);
+    let get_upload = Route::new(Get, "/", get_upload);
+
+    let not_found_catcher = Catcher::new(404, not_found_handler);
+
+    rocket::ignite()
+        .mount("/", vec![always_forward, hello, echo])
+        .mount("/upload", vec![get_upload, post_upload])
+        .mount("/hello", vec![name.clone()])
+        .mount("/hi", vec![name])
+        .catch(vec![not_found_catcher])
+}
+
+fn test(uri: &str, content_type: ContentType, status: Status, body: String) {
+    let mut request = MockRequest::new(Get, uri).header(content_type);
+    let rocket = route_and_mount();
+    let mut response = request.dispatch_with(&rocket);
+
+    assert_eq!(response.status(), status);
+    assert_eq!(response.body().and_then(|b| b.into_string()), Some(body));
+}
+
+#[test]
+fn test_forward() {
+    test("/", ContentType::Plain, Status::Ok, format!("Hello!"));
+}
+
+#[test]
+fn test_name(){
+    let name = "John";
+    let uri = format!("/hello/{}", name);
+    test(&uri, ContentType::Plain, Status::Ok, name.to_string());
+}
+
+#[test]
+fn test_echo() {
+    let echo = "echo text";
+    let uri = format!("/echo:{}", echo);
+    test(&uri, ContentType::Plain, Status::Ok, echo.to_string());
+}
+
+#[test]
+fn test_upload() {
+    let body = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, 
```

Comment:
Can we change to this `expected_body`? I think that will make reading this function a bit easier.